cmd/ctr/introspect.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	gocontext "context"
+	"fmt"
+	"strings"
+
+	"github.com/containerd/containerd/api/services/introspection"
+	"github.com/urfave/cli"
+)
+
+var introspectCommand = cli.Command{
+	Name:  "introspect",
+	Usage: "introspect the daemon",
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "type",
+			Value: "ALL",
+			Usage: "ALL (default) | VERSION | PLUGINS",
+		},
+	},
+	Action: func(context *cli.Context) error {
+		sel := &introspection.IntrospectionSelector{}
+		switch typ := strings.ToUpper(context.String("type")); typ {
+		case "ALL", "":
+			sel.Type = introspection.IntrospectionSelectorType_ALL
```

Comment:
There is a plugin in gogoproto to make these generate correctly. `enum_customname` and `enumvalue_customname` will allow these to follow Go style guidelines for constants. swarmkit has this done correctly in a few places.